### DISCUSSION TITLE
Deck is NOT attached !!

### DISCUSSION THREAD

**Discussion by CalvinKoung** [bitcraze/discussions#838] [Q&A]
Hi, I followed the Step by step:Motion commander instruction. I follow and finish Step 1 and run my python code. It respond Deck is NOT attatched ! but I already attatched the loco position deck and z range deck. 
<img width="734" alt="Screenshot 2023-07-25 at 17 27 24" src="https://github.com/bitcraze/discussions/assets/118436188/40e35eac-1a5e-4649-a7b5-eef378817fc4">
![IMG_1778](https://github.com/bitcraze/discussions/assets/118436188/662c4bd9-047b-4751-bdab-fc452fdba871)
![IMG_1777](https://github.com/bitcraze/discussions/assets/118436188/b9d904ca-a5f8-4796-b03c-1b88acb3e0d6)

**Comment by krichardsson** [comment]
Hi, 
Please reboot the Crazyflie, connect with the python client and copy/paste the contents of the console log tab here so we can take a look at it

**Reply by CalvinKoung** [reply]
Hi, Is this two picture ? 
![IMG_1779](https://github.com/bitcraze/discussions/assets/118436188/249b3af3-bd0a-4c60-bfcd-d8ae274c48e4)
![IMG_1780](https://github.com/bitcraze/discussions/assets/118436188/9eb1af6c-8bfb-4bb6-a2a4-fbf1a30f263a)

**Reply by krichardsson** [reply]
Thanks! 
The one I'm interested in is the first one, but you need to reboot the Crazyflie before you connect to catch the start of the log (which is the important part). So the process is: 
1. reboot the Crayzflie
2. connect with the client
3. Copy/paste the contents of the console log. 

The beginning of the log should look similar to this 
```
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:f83e4269f59b (2023.07) CLEAN
SYS: I am 0x20383347344D50070025003D and I have 1024KB of flash!

... and so on. Around 20-30 lines
```

**Reply by CalvinKoung** [reply]
Hi, the console log can I use python to read out ? Because the client install on company computer and I just happened to be outside these days. Unfortunately, my laptop is m1 macbook cannot install client.

**Reply by krichardsson** [reply]
You can use the [console.py example](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/console/console.py)

**Reply by CalvinKoung** [reply]
Great example, tks.
I saw the two deck_core is OK. 
<img width="482" alt="Screenshot 2023-07-26 at 19 16 18" src="https://github.com/bitcraze/discussions/assets/118436188/d0299dd1-8d5c-463c-9044-8bc88a9ecdf6">

**Reply by krichardsson** [reply]
Thanks!
It looks good, I wanted to see if the decks were initialized properly and looks OK, and they do.

Ah! Now I know what the problem is, I did not read your initial question closely enough, sorry!
The problem is that you don't have a flow deck, you have a Loco deck and Z-ranger deck. The function `param_deck_flow()` checks that there is a flow deck attached and this will fail in your case. The motion commander should work with a loco deck though (if the loco system is set up) so if you want to use the code anyway, you can remove the check that the flow deck is attached.

Just comment out or remove the following lines
```python 
        if not deck_attached_event.wait(timeout=5):
            print('No flow deck detected!')
            sys.exit(1)
```

You will still get the warning that "Deck is NOT attached!" but it should continue to run.

**Reply by CalvinKoung** [reply]
Okay, I got it. So this is test to flow deck only ? That might website have wrong information. 
Thanks your help. 
<img width="901" alt="Screenshot 2023-07-26 at 19 38 51" src="https://github.com/bitcraze/discussions/assets/118436188/3a48c15b-87ce-4c85-9c21-ef24e8ecbf00">

**Reply by krichardsson** [reply]
You are right, that looks wrong. Thanks for letting us know!

**Reply by CalvinKoung** [reply]
No worries, thanks your help.
